autogpt_platform/backend/backend/executor/manager.py

```diff
@@ -952,31 +1018,112 @@ def run_service(self):
         logger.info(f"[{self.service_name}] ⏳ Connecting to Redis...")
         redis.connect()
 
-        sync_manager = multiprocessing.Manager()
+        logger.info(f"[{self.service_name}] Ready to consume messages...")
         while True:
-            graph_exec_data = self.queue.get()
-            graph_exec_id = graph_exec_data.graph_exec_id
-            logger.debug(
-                f"[ExecutionManager] Dispatching graph execution {graph_exec_id}"
+            # cancel graph execution requests
+            method_frame, _, body = channel.basic_get(
+                queue=GRAPH_EXECUTION_CANCEL_QUEUE_NAME,
+                auto_ack=True,
             )
-            cancel_event = sync_manager.Event()
-            future = self.executor.submit(
-                Executor.on_graph_execution, graph_exec_data, cancel_event
+            if method_frame:
+                self._handle_cancel_message(body)
+
+            # start graph execution requests
+            method_frame, _, body = channel.basic_get(
+                queue=GRAPH_EXECUTION_QUEUE_NAME,
+                auto_ack=False,
             )
```

Comment:
are both channels using the saem frame?

Reply:
no, it's just a variable used to read the data